formbricks/formbricks
# feat: cache web builds & playwright binaries in CI

## What does this PR do?
- Defines a composite GitHub Action to introduce Native GitHub Caching for our web builds. Cache Key includes: 
    1. Hash of pnpm lock file
    2. Hash of all (js/ts/jsx/tsx) files in apps/ or packages/ dir
    3. PR number (to currently limit cache sharing per PR/branch) This can be later expanded to across repo
- Uses that in the E2E CI (we can expand further to other tests if this performs well for the next week or so)
- Caches Playwright browser dependencies if hash of lockfile matches (done to make sure none of it/its dependents' version has changed)

## How should this be tested?
- Change something to see the caching getting revalidated
- Make a change in just a github action file and see the build using the cache

PS: For the curious ones, find the my testing playground here: https://github.com/ShubhamPalriwala/formbricks/actions

## Checklist

### Required


### Appreciated
